Ical.Net/CalendarComponents/CalendarEvent.cs

```diff
@@ -42,132 +54,136 @@ public class CalendarEvent : RecurringComponent, IAlarmContainer, IComparable<Ca
+            SetProperty("DURATION", value); 
         }
     }
 
     /// <summary>
-    /// Calculates and returns the duration of the first occurrence of this event.
+    /// Calculates and returns the <b>nominal duration</b> of the first occurrence of this event.
+    /// <para/>
+    /// If the <see cref="CalendarEvent.Duration"/> property is not null, its value will be returned.<br/>
+    /// If <see cref="RecurringComponent.DtStart"/> and <see cref="CalendarEvent.DtEnd"/> are set, it will return <see cref="CalendarEvent.DtEnd"/> minus <see cref="CalendarEvent.DtStart"/>.<br/>
+    /// Otherwise, it will return <see cref="TimeSpan.Zero"/>.
     /// </summary>
     /// <remarks>
-    /// If the 'DURATION' property is set, this method will return its value.
-    /// Otherwise, if DTSTART and DTEND are set, it will return DTSTART minus DTEND.
-    /// Otherwise it will return `default(TimeSpan)`.
-    /// Note that for recurring events, the duration of individual occurrences may vary
-    /// if they span a DST change.
+    /// Note: For recurring events, the <b>effective duration</b> of individual occurrences may vary due to DST transitions
+    /// of the given <see cref="RecurringComponent.DtStart"/> and <see cref="CalendarEvent.DtEnd"/> timezones.
     /// </remarks>
-    /// <returns>The effective duration of this event.</returns>
-    public virtual TimeSpan GetFirstDuration()
+    /// <returns>The <b>nominal duration</b> of this event.</returns>
+    public virtual TimeSpan CalcFirstNominalDuration()
     {
-        if (Properties.ContainsKey("DURATION"))
-            return Duration;
+        // 3.8.5.3. Recurrence Rule
+        // If the duration of the recurring component is specified with the
+        // "DURATION" property, then the same NOMINAL duration will apply to
+        // all the members of the generated recurrence set and the exact
+        // duration of each recurrence instance will depend on its specific
+        // start time.
+        if (Duration is not null)
+            return Duration.Value;
+
+        System.Diagnostics.Debug.Assert(DtStart is not null);
+
+        if (DtEnd is not null)
+        {
+            /*
+                The 'DTEND' property for a 'VEVENT' calendar component specifies the
+                non-inclusive end of the event.
+
+                3.8.5.3. Recurrence Rule:
+                If the duration of the recurring component is specified with the
+                "DTEND" or "DUE" property, then the same EXACT duration will apply
+                to all the members of the generated recurrence set.
+
+                We use the nominal duration here, because the caller will set the period end time to the
+                same timezone as the event end time, finally leading to an exact duration
+                calculation from the zoned start time to the zoned end time.
+             */
+            return DtEnd.Value - DtStart.Value;
```

Comment:
As mentioned above I suggest to go for the exact duration for now (i.e. `DtEnd.AsUtc - DtStart.AsUtc`). *If* calculating the nominal duration, we should throw if `DtEnd.TzId != DtStart.TzId`.